quartodoc/autosummary.py

```diff
@@ -686,14 +698,48 @@ def _generate_sidebar(self, blueprint: layout.Layout):
         if crnt_entry:
             contents.append(crnt_entry)
 
-        entries = [{"id": self.dir, "contents": contents}, {"id": "dummy-sidebar"}]
+        # Create sidebar with user options, ensuring we control `id` and `contents`
+        if self.sidebar is None:
+            sidebar = {}
+        else:
+            sidebar = {k: v for k, v in self.sidebar.items() if k != "file"}
+
+        if "id" not in sidebar:
+            sidebar["id"] = self.dir
+
+        if "contents" not in sidebar:
+            sidebar["contents"] = contents
+        else:
+            if not isinstance(sidebar["contents"], list):
+                raise TypeError("`sidebar.contents` must be a list")
+
+            def splice_contents_recursive(sidebar, contents):
```

Comment:
Do you mind pulling this into a staticmethod on the class (or a function outside the class)? It feels like if it were called `_insert_contents(x: dict | list, contents: ?)` or similar, the idea of a utility function putting contents where a sentinel string is is a bit easier to think about outside of the context of the sidebar.